openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -203,13 +206,18 @@ def from_component(cls, library_key, component):
         """
         Construct a LibraryXBlockMetadata from a Component object.
         """
+        last_publish_log = authoring_api.get_last_publish(component.pk)
+
         return cls(
             usage_key=LibraryUsageLocatorV2(
                 library_key,
                 component.component_type.name,
                 component.local_key,
             ),
             display_name=component.versioning.draft.title,
+            created=component.created,
+            modified=component.versioning.draft.created,
+            last_published=None if last_publish_log is None else last_publish_log.published_at,
```

Comment:
Hi @ormsbee 👋 
I added this code to record the published date for library components in the Studio search index. However, it doesn't work, seemingly because components don't get publish records recorded when they are published -- only the parent library does.

However, there isn't a date available on the `component.versioning.published` ComponentVersion -- the `published.created` matches the `draft.modified` date after publishing, and doesn't represent the actual date published.

What would be the best way to address this issue?

CC @bradenmacdonald

Reply:
cf https://github.com/openedx/edx-platform/pull/35195

Reply:
And https://github.com/openedx/openedx-learning/pull/204

Reply:
I don't remember if there's a good helper for it at the moment, but I think you can get the datetime when a Component was published by following these relations:

Component.publishable_entity -> PublishableEntity.published -> Published.publish_log_record -> PublishLogRecord.publish_log -> PublishLog. published_at